### PR TITLE
Empty states UI Implementation

### PR DESCRIPTION
**Pull request by kokodavid**

### CODE REVIEW

File: app/lib/common/widgets/error_widget.dart [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import 'package:flutter/material.dart';
+
+class ErrorWidgetTemplate extends StatelessWidget {
```
Comment (kumarpalsinh25) [inline app/lib/common/widgets/error_widget.dart:3]:
```suggestion
class EmptyStateWidgetTemplate extends StatelessWidget {
```
- Better to not use `Error word` as `ErrorState` and `EmptyState` are different things.

File: app/lib/features/home/pages/dashboard.dart [hunk 2]
```diff
@@ -54,6 +56,33 @@ class Dashboard extends ConsumerWidget {
...
+      if (children.isEmpty) {
+      return Center(
+        heightFactor: 1.5,
+        child: ErrorWidgetTemplate(
+          title: 'You are not a member of any space yet',
+          subtitle:
+              'Create or join space, to start organizing and collaborating!',
+          image: 'assets/images/empty_home.png',
+          primaryButton: DefaultButton(
+            onPressed: () {},
```
Comment (kumarpalsinh25) [inline app/lib/features/home/pages/dashboard.dart:68]:
```suggestion
            onPressed: () => context.pushNamed(Routes.createSpace.name),
```

File: app/lib/features/chat/widgets/convo_list.dart [hunk 3]
```diff
@@ -43,18 +46,31 @@ class _ConvosListConsumerState extends ConsumerState<ConvosList> {
...
-        child: Text(
-          '${AppLocalizations.of(context)!.loadingConvo}...',
+        heightFactor: 1.5,
+        child: ErrorWidgetTemplate(
+          title: 'You have no DMs at the moment',
+          subtitle:
+              'Get in touch with other change makers, organizers or activists and chat directly with them.',
+          image: 'assets/images/empty_chat.png',
+          primaryButton: DefaultButton(
+            onPressed: () {},
```
Comment (kumarpalsinh25) [inline app/lib/features/chat/widgets/convo_list.dart:58]:
```suggestion
            onPressed: () async => context.pushNamed(
                              Routes.createChat.name,
                            ),
```

File: app/lib/features/home/pages/dashboard.dart [hunk 4]
```diff
@@ -54,6 +56,33 @@ class Dashboard extends ConsumerWidget {
...
+            title: 'Create New Space',
+            style: ElevatedButton.styleFrom(
+              backgroundColor: Theme.of(context).colorScheme.success,
+              disabledBackgroundColor:
+                  Theme.of(context).colorScheme.success.withOpacity(0.5),
+            ),
+          ),
+          secondaryButton: DefaultButton(
+            onPressed: () {},
```
Comment (kumarpalsinh25) [inline app/lib/features/home/pages/dashboard.dart:77]:
```suggestion
            onPressed: () => context.pushNamed(Routes.joinSpace.name),
```

File: app/lib/features/news/widgets/news_widget.dart [hunk 5]
```diff
@@ -32,6 +36,25 @@ class _NewsWidgetState extends ConsumerState<NewsWidget>
...
       data: (data) {
+        if (data.isEmpty) {
+          return Center(
+            child: ErrorWidgetTemplate(
+              title: 'You have no updates',
+              subtitle:
+                  'Create actionable posts and engage everyone within your space.',
+              image: 'assets/images/empty_updates.png',
+              primaryButton: DefaultButton(
+                onPressed: () {},
```
Comment (kumarpalsinh25) [inline app/lib/features/news/widgets/news_widget.dart:47]:
```suggestion
                onPressed: () => context.pushNamed(Routes.actionAddUpdate.name),
```

File: app/lib/features/space/pages/chats_page.dart [hunk 6]
```diff
@@ -115,10 +118,26 @@ class SpaceChatsPage extends ConsumerWidget {
...
-                  heightFactor: 5,
-                  child: Text('Chats are empty'),
+                  heightFactor: 1,
+                  child: ErrorWidgetTemplate(
+                    title: 'No chats in this space yet',
+                    subtitle:
+                        'Get the conversation going to start organizing collaborating',
+                    image: 'assets/images/empty_chat.png',
+                    primaryButton: DefaultButton(
+                      onPressed: () {},
```
Comment (kumarpalsinh25) [inline app/lib/features/space/pages/chats_page.dart:130]:
```suggestion
                      onPressed: () => context.pushNamed(
                            Routes.createChat.name,
                            queryParameters: {'spaceId': spaceIdOrAlias},
                            extra: 1,
                          ),```

File: app/lib/features/space/pages/events_page.dart [hunk 7]
```diff
@@ -52,10 +55,26 @@ class SpaceEventsPage extends ConsumerWidget {
...
-                    child: Text(
-                      'Currently there are no events planned for this space',
+                    heightFactor: 1,
+                    child: ErrorWidgetTemplate(
+                      title: 'No events planned yet',
+                      subtitle:
+                          'Create new event and bring your community together',
+                      image: 'assets/images/empty_events.png',
+                      primaryButton: DefaultButton(
+                        onPressed: () {},
```
Comment (kumarpalsinh25) [inline app/lib/features/space/pages/events_page.dart:67]:
```suggestion
                    onPressed: () => context.pushNamed(Routes.createEvent.name),
```

File: app/lib/features/space/pages/chats_page.dart [hunk 8]
```diff
@@ -115,10 +118,26 @@ class SpaceChatsPage extends ConsumerWidget {
...
                 child: Center(
-                  heightFactor: 5,
-                  child: Text('Chats are empty'),
+                  heightFactor: 1,
+                  child: ErrorWidgetTemplate(
+                    title: 'No chats in this space yet',
+                    subtitle:
+                        'Get the conversation going to start organizing collaborating',
+                    image: 'assets/images/empty_chat.png',
+                    primaryButton: DefaultButton(
```
Comment (kumarpalsinh25) [inline app/lib/features/space/pages/chats_page.dart:129]: `Create Space Chat` should be based on the permission `CanLinkSpaces`

File: app/lib/features/space/pages/events_page.dart [hunk 9]
```diff
@@ -52,10 +55,26 @@ class SpaceEventsPage extends ConsumerWidget {
...
                   child: Center(
-                    child: Text(
-                      'Currently there are no events planned for this space',
+                    heightFactor: 1,
+                    child: ErrorWidgetTemplate(
+                      title: 'No events planned yet',
+                      subtitle:
+                          'Create new event and bring your community together',
+                      image: 'assets/images/empty_events.png',
+                      primaryButton: DefaultButton(
```
Comment (kumarpalsinh25) [inline app/lib/features/space/pages/events_page.dart:66]: `Create Event` should be based on the permission `CanPostEvent`

**Comment by gnunicorn** [comment]
@gtalha07 mind providing screenshots?

File: .changes/1231-empty-states.dart [hunk 10]
```diff
@@ -0,0 +1 @@
+- screens now have empty states when there is no data to be displayed.
```
Comment (gnunicorn) [inline .changes/1231-empty-states.dart:1]:
```suggestion
- Screens without data to show now present a helpful text
```

File: app/lib/common/themes/app_theme.dart [hunk 11]
```diff
@@ -1,4 +1,4 @@
-import 'package:flutter/material.dart';
+ import 'package:flutter/material.dart';
```
Comment (gnunicorn) [inline app/lib/common/themes/app_theme.dart:1]:
```suggestion
import 'package:flutter/material.dart';
```

mistake I am sure.

File: app/lib/features/space/pages/pins_page.dart [hunk 12]
```diff
@@ -61,9 +64,27 @@ class SpacePinsPage extends ConsumerWidget {
...
+                return  SliverToBoxAdapter(
                   child: Center(
-                    child: Text('there is nothing pinned yet'),
+                    heightFactor: 1,
+                    child: ErrorWidgetTemplate(
+                      title: 'No pins available yet',
+                      subtitle:
+                          'Share important resources with your community such as documents or links so everyone is updated.',
+                      image: 'assets/images/empty_pin.png',
+                      primaryButton: DefaultButton(
```
Comment (kumarpalsinh25) [inline app/lib/features/space/pages/pins_page.dart:75]: `Share Pin` should be based on the permission `CanPostPin`

File: app/lib/features/space/pages/pins_page.dart [hunk 13]
```diff
@@ -61,9 +64,27 @@ class SpacePinsPage extends ConsumerWidget {
...
+                        onPressed: () {},
```
Comment (kumarpalsinh25) [inline app/lib/features/space/pages/pins_page.dart:76]:
```suggestion
                        onPressed: () => context.pushNamed(
                      Routes.actionAddPin.name,
                      queryParameters: {'spaceId': spaceIdOrAlias},
                    ),
```

File: app/lib/common/widgets/error_widget.dart (hunk 1 above)
Comment (gnunicorn) [inline app/lib/common/widgets/error_widget.dart:3]: As this is meant to show the empty state - which is a legitimate state - this shouldn't be named "Error"Widget ... It's the NoDataToShow or EmptyState or something...

File: app/lib/features/space/pages/related_spaces_page.dart [hunk 14]
```diff
@@ -254,8 +256,33 @@ class RelatedSpacesPage extends ConsumerWidget {
...
+                items.add(
+                  SliverToBoxAdapter(
+                    child: Center(
+                      heightFactor: 1,
+                      child: ErrorWidgetTemplate(
+                        title: 'No connected spaces',
+                        subtitle:
+                            'In connected spaces, you can focus on specific actions or campaigns of your working groups and start organizing.',
+                        image: 'assets/images/empty_space.png',
+                        primaryButton: DefaultButton(
```
Comment (kumarpalsinh25) [inline app/lib/features/space/pages/related_spaces_page.dart:270]: `Create New Spaces` should be based on the permission `CanLinkSpaces`

File: app/lib/features/chat/widgets/convo_list.dart (hunk 3 above)
Comment (gnunicorn) [inline app/lib/features/chat/widgets/convo_list.dart:58]: Pretty sure we agreed to not add buttons without function anymore. Please either remove the button for now or better route to the actual activity.

File: app/lib/features/space/pages/related_spaces_page.dart [hunk 15]
```diff
@@ -254,8 +256,33 @@ class RelatedSpacesPage extends ConsumerWidget {
...
+                          onPressed: () {},
```
Comment (kumarpalsinh25) [inline app/lib/features/space/pages/related_spaces_page.dart:271]:
```suggestion
                          onPressed: () => context.pushNamed(
                            Routes.createSpace.name,
                            queryParameters: {'parentSpaceId': spaceIdOrAlias},
                          ),,
```

File: app/lib/features/home/pages/dashboard.dart (hunk 2 above)
Comment (gnunicorn) [inline app/lib/features/home/pages/dashboard.dart:68]: same non-func-button again.

File: app/lib/features/news/widgets/news_widget.dart (hunk 5 above)
Comment (gnunicorn) [inline app/lib/features/news/widgets/news_widget.dart:47]: same non-func-button.

**Review by gnunicorn — CHANGES_REQUESTED** [review]
I stopped the review after the third non-functional button being added. Please either give these buttons the appropriate functionality or remove them from the view...

**Review by kumarpalsinh25 — CHANGES_REQUESTED** [review]
Looks good overall. 
Few typo suggestions, missing button click events and permission checking stuff need to be handled before merge.

Also if you can share Screenshots then it will be better to review the design work.

File: app/lib/common/widgets/error_widget.dart (hunk 1 above)
Comment (kokodavid) [reply app/lib/common/widgets/error_widget.dart:3]: Renaming it to EmptyState

**Comment by kokodavid** [comment]
> Looks good overall. Few typo suggestions, missing button click events and permission checking stuff need to be handled before merge.
> 
> Also if you can share Screenshots then it will be better to review the design work.

In the issue, I have provided all screenshots for the states implemented. Would you suggest I share the screenshots in the PR instead?

**Comment by kumarpalsinh25** [comment]
> In the issue, I have provided all screenshots for the states implemented. Would you suggest I share the screenshots in the PR instead?

@kokodavid 
All the good with the Figma Design, I was asking for the screenshot after implementation on Flutter to cross check the UI.

File: app/lib/features/chat/widgets/convo_list.dart (hunk 3 above)
Comment (kokodavid) [reply app/lib/features/chat/widgets/convo_list.dart:58]: Fixed the routing issues for the non functional buttons .

**Comment by kokodavid** [comment]
> > In the issue, I have provided all screenshots for the states implemented. Would you suggest I share the screenshots in the PR instead?
> 
> @kokodavid All the good with the Figma Design, I was asking for the screenshot after implementation on Flutter to cross check the UI.

The Screenshots for implementation on flutter are all linked here @kumarpalsinh25 [https://github.com/acterglobal/a3/issues/1198](url)

File: app/assets/images/empty_activities.png
[the remark this answers is not in the document]
Comment (kokodavid) [reply app/assets/images/empty_activities.png:1]: @gtalha07 @kumarpalsinh25 I have tried to add them as SVG and this is how they appear, some of the content is missing. 
image

[the remark this answers is not in the document]
Comment (gtalha07) [reply app/assets/images/empty_activities.png:1]: @kokodavid  Maybe clean the svg from any online tool before adding it ? It gets noisy sometimes.

**Review by kumarpalsinh25 — APPROVED** [review]
LGTM!

**Review by gtalha07 — APPROVED** [review]
Great!

**Comment by gnunicorn** [comment]
what is keeping this from being merged?

**Review by gnunicorn — APPROVED** [review]

**Pull request merged**